shm/alcohol.cpp

```diff
@@ -42,4 +42,8 @@ bool Alcohol::operator==(const Cargo& other) const {
 void Alcohol::nextDay(size_t elapsedTime)
 {
     basePrice_ += (elapsedTime * 2);
+}
+Cargo* Alcohol::clone()
+{
+    return new Alcohol(*this);
```

Comment:
Avoid raw ptr! and new operator

return std::unique_ptr<Cargo> and use std::make_unique.